packages/plugin-commands-setup/src/setupOnWindows.ts

```diff
@@ -29,9 +30,21 @@ function pathIncludesDir (pathValue: string, dir: string): boolean {
 export async function setupWindowsEnvironmentPath (pnpmHomeDir: string, opts: { force: boolean }): Promise<string> {
   // Use `chcp` to make `reg` use utf8 encoding for output.
   // Otherwise, the non-ascii characters in the environment variables will become garbled characters.
-  const queryResult = await execa(`chcp 65001>nul && reg query ${REG_KEY}`, undefined, { shell: true })
+
+  const chcpResult = await execa('chcp')
+  const cpMatch = /\d+/.exec(chcpResult.stdout) ?? []
+  const cpBak = parseInt(cpMatch[0])
+  if (chcpResult.failed || cpBak === 0) {
+    return `exec chcp failed: ${cpBak}, ${chcpResult.stderr}`
+  }
+
```

Comment:
Oh! `chcpResult.failed && !(cpBak > 0)` should be used here.

Because the result may be `NaN`, although this seems unlikely.